### PR TITLE
chore(client): Migration of SolutionViewer component to Typescript.

### LINKED ISSUE #42256: TypeScript migration
## Co-ordination

As discussed in the last contributors [meeting](https://forum.freecodecamp.org/t/office-hour-contributors-may-13-2021/458257/4) we're looking to migrate our client to TypeScript. 

In order to help people work in parallel this issue can be used to track who is working on what. I've put a checklist of all the unmigrated files at the bottom.  If you want to work on a file or some files, please edit in your github username next to the file(s) you want to work on.  If you don't have edit permission, please just reply to this issue @ me and I'll take care of it.

### Small is beautiful

If a file can be migrated on its own, that's perfect.  That makes the PR incredibly easy to review and minimises the chance we break anything.

### PropTypes can go

None of the components in the client dirs are intended to be consumed by JavaScript components - eventually everything will be `tsx`.  With that in mind static type checking should be enough.

### Ignore linting if necessary

Migration should be incremental and painless.  If you find that converting to TS requires a big rewrite, feel free to `@ts-ignore` or `eslint-disable` as needed.

It's more important that each change is easy to understand and review than it is it write perfect TS.  We'll get there eventually.

### More guidelines may get added

As we do this, chances are we'll revise the approach.  Any thoughts/comments/critiques are welcome, but I wanted to keep this straightforward for now.

<details> <summary> list of files left to migrate (collapsed because it's rather long) </summary>

- [ ] /client/utils/buildChallenges.js
- [ ] /client/utils/challengeTypes.js
- [ ] /client/utils/index.js
- [ ] /client/utils/stateManagement.js
- [ ] /client/utils/tags.js
- [ ] /client/utils/tags.test.js
- [ ] /client/utils/gatsby/challengePageCreator.js
- [ ] /client/utils/gatsby/index.js
- [ ] /client/utils/gatsby/layoutSelector.js
- [ ] /client/utils/gatsby/layoutSelector.test.js
- [ ] /client/src/html.js
- [ ] /client/src/utils/ajax.js
- [ ] /client/src/utils/algolia-locale-setup.js
- [ ] /client/src/utils/createTypes.js
- [ ] /client/src/utils/curriculum-helpers.test.js
- [ ] /client/src/utils/format.js
- [ ] /client/src/utils/format.test.js
- [ ] /client/src/utils/get-words.js
- [ ] /client/src/utils/handled-error.js
- [ ] /client/src/utils/handled-error.test.js
- [ ] /client/src/utils/index.js
- [ ] /client/src/utils/reallyWeirdErrorMessage.js
- [ ] /client/src/utils/report-error.js
- [ ] /client/src/utils/reportedErrorMessage.js
- [ ] /client/src/utils/scriptLoaders.js
- [ ] /client/src/utils/standardErrorMessage.js
- [ ] /client/src/utils/to-learn-path.js
- [ ] /client/src/utils/__fixtures/curriculum-helpers-css.js
- [ ] /client/src/utils/__fixtures/curriculum-helpers-html.js
- [ ] /client/src/utils/__fixtures/curriculum-helpers-javascript.js
- [ ] /client/src/utils/__fixtures/curriculum-helpers-remove-white-space.js
- [ ] /client/src/templates/Introduction/Intro.js
- [ ] /client/src/templates/Introduction/SuperBlockIntro.js
- [ ] /client/src/templates/Introduction/redux/index.js
- [ ] /client/src/templates/Introduction/components/Block.js
- [ ] /client/src/templates/Introduction/components/CertChallenge.js
- [ ] /client/src/templates/Introduction/components/CertificationCard.js
- [ ] /client/src/templates/Introduction/components/Challenges.js
- [ ] /client/src/templates/Introduction/components/ClaimCertSteps.js
- [ ] /client/src/templates/Introduction/components/SuperBlockIntro.js
- [ ] /client/src/templates/Challenges/video/Show.js
- [ ] /client/src/templates/Challenges/utils/build.js
- [ ] /client/src/templates/Challenges/utils/fetch-and-cache.js
- [ ] /client/src/templates/Challenges/utils/frame.js
- [ ] /client/src/templates/Challenges/utils/getTargetEditor.js
- [ ] /client/src/templates/Challenges/utils/index.js
- [ ] /client/src/templates/Challenges/utils/index.test.js
- [ ] /client/src/templates/Challenges/utils/postUpdate$.js
- [ ] /client/src/templates/Challenges/utils/worker-executor.js
- [ ] /client/src/templates/Challenges/utils/worker-executor.test.js
- [ ] /client/src/templates/Challenges/redux/code-lock-epic.js
- [ ] /client/src/templates/Challenges/redux/code-storage-epic.js
- [ ] /client/src/templates/Challenges/redux/completion-epic.js
- [ ] /client/src/templates/Challenges/redux/create-question-epic.js
- [ ] /client/src/templates/Challenges/redux/create-question-epic.test.js
- [ ] /client/src/templates/Challenges/redux/current-challenge-saga.js
- [ ] /client/src/templates/Challenges/redux/execute-challenge-saga.js
- [ ] /client/src/templates/Challenges/redux/index.js
- [ ] /client/src/templates/Challenges/rechallenge/builders.js
- [ ] /client/src/templates/Challenges/rechallenge/throwers.js
- [ ] /client/src/templates/Challenges/rechallenge/transformers.js
- [ ] /client/src/templates/Challenges/projects/SolutionForm.js
- [ ] /client/src/templates/Challenges/projects/Tool-Panel.js
- [ ] /client/src/templates/Challenges/projects/frontend/Show.js
- [ ] /client/src/templates/Challenges/projects/backend/Show.js
- [ ] /client/src/templates/Challenges/components/BreadCrumb.js
- [ ] /client/src/templates/Challenges/components/Challenge-Description.js
- [ ] /client/src/templates/Challenges/components/Challenge-Title.js
- [ ] /client/src/templates/Challenges/components/ChallengeTitle.test.js
- [ ] /client/src/templates/Challenges/components/CompletionModal.js
- [ ] /client/src/templates/Challenges/components/CompletionModal.test.js
- [ ] /client/src/templates/Challenges/components/CompletionModalBody.js
- [ ] /client/src/templates/Challenges/components/CompletionModalBody.test.js
- [ ] /client/src/templates/Challenges/components/HelpModal.js
- [ ] /client/src/templates/Challenges/components/Hotkeys.js
- [ ] /client/src/templates/Challenges/components/Output.js
- [ ] /client/src/templates/Challenges/components/Preview.js
- [ ] /client/src/templates/Challenges/components/PrismFormatted.js
- [ ] /client/src/templates/Challenges/components/ResetModal.js
- [ ] /client/src/templates/Challenges/components/Side-Panel.js
- [ ] /client/src/templates/Challenges/components/Test-Suite.js
- [ ] /client/src/templates/Challenges/components/Tool-Panel.js
- [ ] /client/src/templates/Challenges/components/VideoModal.js
- [ ] /client/src/templates/Challenges/classic/ActionRow.js
- [ ] /client/src/templates/Challenges/classic/DesktopLayout.js
- [ ] /client/src/templates/Challenges/classic/Editor.js
- [ ] /client/src/templates/Challenges/classic/EditorTabs.js
- [ ] /client/src/templates/Challenges/classic/MobileLayout.js
- [ ] /client/src/templates/Challenges/classic/MultifileEditor.js
- [ ] /client/src/templates/Challenges/classic/Show.js
- [ ] /client/src/resources/certAndProjectMap.js
- [ ] /client/src/resources/honesty-policy.js
- [ ] /client/src/redux/accept-terms-saga.js
- [ ] /client/src/redux/app-mount-saga.js
- [ ] /client/src/redux/cookieValues.js
- [ ] /client/src/redux/createStore.js
- [ ] /client/src/redux/donation-saga.js
- [ ] /client/src/redux/error-saga.js
- [ ] /client/src/redux/failed-updates-epic.js
- [ ] /client/src/redux/failed-updates-epic.test.js
- [ ] /client/src/redux/fetch-user-saga.js
- [ ] /client/src/redux/ga-saga.js
- [ ] /client/src/redux/ga-saga.test.js
- [ ] /client/src/redux/hard-go-to-epic.js
- [ ] /client/src/redux/index.js
- [ ] /client/src/redux/night-mode-saga.js
- [ ] /client/src/redux/propTypes.js
- [ ] /client/src/redux/report-user-saga.js
- [ ] /client/src/redux/rootEpic.js
- [ ] /client/src/redux/rootReducer.js
- [ ] /client/src/redux/rootSaga.js
- [ ] /client/src/redux/show-cert-saga.js
- [ ] /client/src/redux/update-complete-epic.js
- [ ] /client/src/redux/settings/danger-zone-saga.js
- [ ] /client/src/redux/settings/index.js
- [ ] /client/src/redux/settings/settings-sagas.js
- [ ] /client/src/redux/settings/update-email-saga.js
- [ ] /client/src/redux/settings/update-legacy-certificate-saga.js
- [ ] /client/src/pages/404.js
- [ ] /client/src/pages/certification.js
- [ ] /client/src/pages/challenges.js
- [ ] /client/src/pages/challenges.test.js
- [ ] /client/src/pages/donate.js
- [ ] /client/src/pages/email-sign-up.js
- [ ] /client/src/pages/index.js
- [ ] /client/src/pages/learn.js
- [ ] /client/src/pages/search.js
- [ ] /client/src/pages/settings.js
- [ ] /client/src/pages/unsubscribed.js
- [ ] /client/src/pages/update-email.js
- [ ] /client/src/pages/user.js
- [ ] /client/src/components/AppMountNotifier.js
- [ ] /client/src/components/RedirectHome.js
- [ ] /client/src/components/createExternalRedirects.js
- [ ] /client/src/components/createExternalRedirects.test.js
- [ ] /client/src/components/createLanguageRedirect.js
- [ ] /client/src/components/createLanguageRedirect.test.js
- [ ] /client/src/components/createRedirect.js
- [ ] /client/src/components/settings/About.js
- [ ] /client/src/components/settings/Certification.js
- [ ] /client/src/components/settings/Certification.test.js
- [ ] /client/src/components/settings/DangerZone.js
- [ ] /client/src/components/settings/DeleteModal.js
- [ ] /client/src/components/settings/Email.js
- [ ] /client/src/components/settings/Honesty.js
- [ ] /client/src/components/settings/Honesty.test.js
- [ ] /client/src/components/settings/Internet.js
- [ ] /client/src/components/settings/Portfolio.js
- [ ] /client/src/components/settings/Privacy.js
- [ ] /client/src/components/settings/ResetModal.js
- [ ] /client/src/components/settings/SectionHeader.js
- [ ] /client/src/components/settings/Theme.js
- [ ] /client/src/components/settings/ToggleSetting.js
- [ ] /client/src/components/settings/Username.js
- [ ] /client/src/components/search/WithInstantSearch.js
- [ ] /client/src/components/search/searchPage/EmptySearch.js
- [ ] /client/src/components/search/searchPage/NoResults.js
- [ ] /client/src/components/search/searchPage/SearchPageHits.js
- [ ] /client/src/components/search/searchBar/NoHitsSuggestion.js
- [ ] /client/src/components/search/searchBar/SearchBar.js
- [ ] /client/src/components/search/searchBar/SearchBar.test.js
- [ ] /client/src/components/search/searchBar/SearchHits.js
- [ ] /client/src/components/search/searchBar/SearchSuggestion.js
- [ ] /client/src/components/search/redux/index.js
- [ ] /client/src/components/profile/Profile.js
- [ ] /client/src/components/profile/Profile.test.js
- [ ] /client/src/components/profile/components/Camper.js
- [ ] /client/src/components/profile/components/Certifications.js
- [ ] /client/src/components/profile/components/HeatMap.js
- [ ] /client/src/components/profile/components/HeatMap.test.js
- [ ] /client/src/components/profile/components/Portfolio.js
- [ ] /client/src/components/profile/components/SocialIcons.js
- [ ] /client/src/components/profile/components/TimeLine.js
- [ ] /client/src/components/profile/components/TimeLine.test.js
- [ ] /client/src/components/profile/components/TimelinePagination.js
- [ ] /client/src/components/layouts/Certification.js
- [ ] /client/src/components/layouts/Default.js
- [ ] /client/src/components/layouts/Learn.js
- [ ] /client/src/components/layouts/index.js
- [ ] /client/src/components/landing/Landing.test.js
- [ ] /client/src/components/landing/index.js
- [ ] /client/src/components/landing/components/AsSeenIn.js
- [ ] /client/src/components/landing/components/BigCallToAction.js
- [ ] /client/src/components/landing/components/CampersImage.js
- [ ] /client/src/components/landing/components/Certifications.js
- [ ] /client/src/components/landing/components/LandingTop.js
- [ ] /client/src/components/landing/components/Testimonials.js
- [ ] /client/src/components/helpers/AvatarRenderer.js
- [ ] /client/src/components/helpers/ButtonSpacer.js
- [ ] /client/src/components/helpers/CurrentChallengeLink.js
- [ ] /client/src/components/helpers/FullWidthRow.js
- [ ] /client/src/components/helpers/ImageLoader.js
- [ ] /client/src/components/helpers/Link.js
- [ ] /client/src/components/helpers/Link.test.js
- [ ] /client/src/components/helpers/Loader.js
- [ ] /client/src/components/helpers/Loader.test.js
- [ ] /client/src/components/helpers/SkeletonSprite.js
- [ ] /client/src/components/helpers/SlimWidthRow.js
- [ ] /client/src/components/helpers/Spacer.js
- [ ] /client/src/components/helpers/ToggleButton.js
- [ ] /client/src/components/helpers/borderColorPicker.js
- [ ] /client/src/components/helpers/index.js
- [ ] /client/src/components/helpers/skeletonStyles.js
- [ ] /client/src/components/helpers/form/BlockSaveButton.js
- [ ] /client/src/components/helpers/form/BlockSaveWrapper.js
- [ ] /client/src/components/formHelpers/BlockSaveButton.js
- [ ] /client/src/components/formHelpers/BlockSaveButton.test.js
- [ ] /client/src/components/formHelpers/BlockSaveWrapper.js
- [ ] /client/src/components/formHelpers/BlockSaveWrapper.test.js
- [ ] /client/src/components/formHelpers/Form.js
- [ ] /client/src/components/formHelpers/Form.test.js
- [ ] /client/src/components/formHelpers/FormFields.js
- [ ] /client/src/components/formHelpers/FormValidators.js
- [ ] /client/src/components/formHelpers/index.js
- [ ] /client/src/components/SolutionViewer/ProjectModal.js
- [ ] /client/src/components/SolutionViewer/SolutionViewer.js
- [ ] /client/src/components/OfflineWarning/OfflineWarning.js
- [ ] /client/src/components/OfflineWarning/index.js
- [ ] /client/src/components/Map/Map.test.js
- [ ] /client/src/components/Map/index.js
- [ ] /client/src/components/Intro/Intro.test.js
- [ ] /client/src/components/Intro/index.js
- [ ] /client/src/components/Intro/components/IntroDescription.js
- [ ] /client/src/components/Header/Header.test.js
- [ ] /client/src/components/Header/index.js
- [ ] /client/src/components/Header/components/AuthOrProfile.js
- [ ] /client/src/components/Header/components/Login.js
- [ ] /client/src/components/Header/components/MenuButton.js
- [ ] /client/src/components/Header/components/NavLinks.js
- [ ] /client/src/components/Header/components/NavLogo.js
- [ ] /client/src/components/Header/components/UniversalNav.js
- [ ] /client/src/components/Header/components/UserState.js
- [ ] /client/src/components/FourOhFour/index.js
- [ ] /client/src/components/Footer/Footer.test.js
- [ ] /client/src/components/Footer/index.js
- [ ] /client/src/components/Flash/redux/index.js
- [ ] /client/src/components/Donation/DonateCompletion.js
- [ ] /client/src/components/Donation/DonateForm.js
- [ ] /client/src/components/Donation/DonationModal.js
- [ ] /client/src/components/Donation/DonationTextComponents.js
- [ ] /client/src/components/Donation/PayPalButtonScriptLoader.js
- [ ] /client/src/components/Donation/PaypalButton.js
- [ ] /client/src/components/Donation/assets/ApplePay.js
- [ ] /client/src/components/Donation/assets/GooglePay.js
- [ ] /client/src/client-only-routes/ShowCertification.js
- [ ] /client/src/client-only-routes/ShowProfileOrFourOhFour.js
- [ ] /client/src/client-only-routes/ShowProjectLinks.js
- [ ] /client/src/client-only-routes/ShowSettings.js
- [ ] /client/src/client-only-routes/ShowSettings.test.js
- [ ] /client/src/client-only-routes/ShowUnsubscribed.js
- [ ] /client/src/client-only-routes/ShowUser.js
- [ ] /client/src/client/frame-runner.js
- [ ] /client/src/client/workers/sass-compile.js
- [ ] /client/src/client/workers/test-evaluator.js
- [ ] /client/src/assets/images/components/AmazonLogo.js
- [ ] /client/src/assets/images/components/AppleLogo.js
- [ ] /client/src/assets/images/components/AsFeatureLogo.js
- [ ] /client/src/assets/images/components/GoogleLogo.js
- [ ] /client/src/assets/images/components/MicrosoftLogo.js
- [ ] /client/src/assets/images/components/SpotifyLogo.js
- [ ] /client/src/assets/images/components/index.js
- [ ] /client/src/assets/icons/APIIcon.js
- [ ] /client/src/assets/icons/Algorithm.js
- [ ] /client/src/assets/icons/Analytics.js
- [ ] /client/src/assets/icons/Caret.js
- [ ] /client/src/assets/icons/CertificationIcon.js
- [ ] /client/src/assets/icons/Clipboard.js
- [ ] /client/src/assets/icons/Cup.js
- [ ] /client/src/assets/icons/D3Icon.js
- [ ] /client/src/assets/icons/DefaultAvatar.js
- [ ] /client/src/assets/icons/DonateWithPayPal.js
- [ ] /client/src/assets/icons/Fail.js
- [ ] /client/src/assets/icons/FreeCodeCampLogo.js
- [ ] /client/src/assets/icons/GreenNotCompleted.js
- [ ] /client/src/assets/icons/GreenPass.js
- [ ] /client/src/assets/icons/Heart.js
- [ ] /client/src/assets/icons/Initial.js
- [ ] /client/src/assets/icons/IntroInformation.js
- [ ] /client/src/assets/icons/JavaScriptIcon.js
- [ ] /client/src/assets/icons/LinkButton.js
- [ ] /client/src/assets/icons/PythonIcon.js
- [ ] /client/src/assets/icons/ReactIcon.js
- [ ] /client/src/assets/icons/ResponsiveDesign.js
- [ ] /client/src/assets/icons/Shield.js
- [ ] /client/src/assets/icons/Spacer.js
- [ ] /client/src/assets/icons/TensorflowIcon.js
- [ ] /client/src/assets/icons/ToggleCheck.js
- [ ] /client/src/assets/icons/index.js
- [ ] /client/src/analytics/index.js
- [ ] /client/src/analytics/__mocks__/index.js
- [ ] /client/src/__tests__/integration/handled-error.test.js
- [ ] /client/src/__mocks__/challenge-nodes.js
- [ ] /client/src/__mocks__/fileMock.js
- [ ] /client/src/__mocks__/gatsby-link.js
- [ ] /client/src/__mocks__/gatsby.js
- [ ] /client/src/__mocks__/react-i18nextMock.js
- [ ] /client/src/__mocks__/styleMock.js
- [ ] /client/plugins/gatsby-remark-node-identity/gatsby-node.js
- [ ] /client/plugins/fcc-source-challenges/create-challenge-nodes.js
- [ ] /client/plugins/fcc-source-challenges/gatsby-node.js
- [ ] /client/i18n/config.js
- [ ] /client/i18n/configForTests.js
- [ ] /client/i18n/locales.test.js
- [ ] /client/i18n/schema-validation.js
- [ ] /client/i18n/validate-keys.js

</details>

### PR DESCRIPTION
**Pull request by ashiskumar-1999**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Reference: #42256

### CODE REVIEW

**Comment by ieahleen** [comment]
Hey @ashiskumar-1999 

We would love to be able to merge your changes but it looks like there is an error with the CI build. ⚠️

Once you resolve these issues, we will be able to review your PR and merge it. 😊

---

Feel free to reference the [contributing guidelines](https://contribute.freecodecamp.org/#/how-to-work-on-coding-challenges?id=testing-challenges) for instructions on running the CI build locally. ✅

**Comment by ashiskumar-1999** [comment]
Hey @ieahleen, I have run the test command locally and it passed without any errors. Can you tell me specifically about the issues that you have mentioned.

**Comment by ieahleen** [comment]
Sorry, it failed on files you did not change

**Comment by nhcarrigan** [comment]
@ieahleen @ashiskumar-1999 the errors in the checks are TypeScript compile errors. Because the `ProjectModal` was converted to TypeScript, the `show-project-links` is throwing compile errors due to the new ProjectModal type definitions not matching.

Give me a few moments and I'll propose some fixes. 🙂

File: client/src/components/SolutionViewer/ProjectModal.tsx
Comment (nhcarrigan) [inline client/src/components/SolutionViewer/ProjectModal.tsx:53]: I can't reach the line because of GitHub restrictions, but above here on line 47:

Replace
```tsx
        <SolutionViewer challengeFiles={challengeFiles} solution={solution} />
```
with
```tsx
        <SolutionViewer
          challengeFiles={challengeFiles}
          solution={String(solution)}
        />
```

Going to cc @ojeytonwilliams on this - it's a cheaty approach to address the type def, but might be better to re-evaluate the type def?

File: client/src/components/SolutionViewer/SolutionViewer.tsx
Comment (nhcarrigan) [inline client/src/components/SolutionViewer/SolutionViewer.tsx:4]:
```suggestion
import React from 'react';
import { ChallengeFile, ChallengeFiles } from '../../redux/prop-types';
```
Necessary import

File: client/src/components/SolutionViewer/SolutionViewer.tsx
Comment (nhcarrigan) [inline client/src/components/SolutionViewer/SolutionViewer.tsx:7]:
```suggestion
  challengeFiles: ChallengeFiles;
```
Proper TypeDef

File: client/src/components/SolutionViewer/SolutionViewer.tsx
Comment (nhcarrigan) [inline client/src/components/SolutionViewer/SolutionViewer.tsx:18]:
```suggestion
        challengeFiles.map((challengeFile: ChallengeFile) => (
```
This is not a string.

File: client/src/components/SolutionViewer/SolutionViewer.tsx
Comment (nhcarrigan) [inline client/src/components/SolutionViewer/SolutionViewer.tsx:32]:
```suggestion
                      challengeFile.contents.trim(),
                      Prism.languages[challengeFile.ext],
```
Correct property access.

File: client/src/components/SolutionViewer/SolutionViewer.tsx
Comment (nhcarrigan) [inline client/src/components/SolutionViewer/SolutionViewer.tsx:24]:
```suggestion
            <Panel.Heading>{challengeFile.ext.toUpperCase()}</Panel.Heading>
```
Correct property access.

File: client/src/components/SolutionViewer/ProjectModal.tsx
Comment (nhcarrigan) [inline client/src/components/SolutionViewer/ProjectModal.tsx:9]: This comment can be removed now.

File: client/src/components/SolutionViewer/SolutionViewer.tsx
Comment (nhcarrigan) [inline client/src/components/SolutionViewer/SolutionViewer.tsx:6]:
```suggestion
type SolutionViewerProps = {
```
PascalCase, also don't believe this needs to be an interface.

File: client/src/components/SolutionViewer/SolutionViewer.tsx
Comment (nhcarrigan) [inline client/src/components/SolutionViewer/SolutionViewer.tsx:14]:
```suggestion
}: SolutionViewerProps): JSX.Element {
```
PascalCase

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
These changes should resolve the TypeScript compile errors (unless I missed one), and allow us to move forward with the review process. 🙂

**Comment by ashiskumar-1999** [comment]
Hey @nhcarrigan, Have you done the changes in the pull request or just suggest doing it?
Because when I see the commits, your changes are in the first. So, I am confused a little bit.

**Comment by ojeytonwilliams** [comment]
@ashiskumar-1999 they're just suggestions. If you look at https://github.com/freeCodeCamp/freeCodeCamp/pull/43694/commits you can see there's only your original commit.

**Comment by ashiskumar-1999** [comment]
Thank you @ojeytonwilliams, I am changing the code accordingly and push it to the pull request.

**Comment by ieahleen** [comment]
@ashiskumar-1999 you can commit from the suggestions, go to /files, then use the button "Add suggestion to batch", and once you have finished at the top there is a "Commit" button that will let you commit all the suggestions

**Comment by ashiskumar-1999** [comment]
Hey @ieahleen @ojeytonwilliams, I have made the changes in my commits. You can now review it and merge it if there will be no more errors.

**Comment by ashiskumar-1999** [comment]
Hey @nhcarrigan, I have added all the suggestions but this error came up. 
Error: client/src/components/SolutionViewer/SolutionViewer.tsx(46,16): error TS2531: Object is possibly 'null'.
Can you help me to identify this issue?

File: client/src/components/SolutionViewer/ProjectModal.tsx
Comment (nhcarrigan) [reply client/src/components/SolutionViewer/ProjectModal.tsx:54]: image

Looks like this hasn't been changed yet.

File: client/src/components/SolutionViewer/SolutionViewer.tsx
Comment (nhcarrigan) [inline client/src/components/SolutionViewer/SolutionViewer.tsx:9]:
```suggestion
  solution: string;
```
This one should stay a string - when you handle the fix in the other comment, the error for this being possibly null/undefined will go away.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Couple of steps should fix that error.

**Comment by ashiskumar-1999** [comment]
Hey @ieahleen, Can you now review it?
I think there is no error left to be resolved but there are two checks pending.

**Comment by gikf** [comment]
> I think there is no error left to be resolved but there are two checks pending.

I've poked them for you :slightly_smiling_face:

**Comment by ashiskumar-1999** [comment]
Hey @gikf, Can we ignore this prettier error?

**Comment by gikf** [comment]
> Hey @gikf, Can we ignore this prettier error? When I did some changes yesterday, the SolutionViewer.tsx had no prettier issue but now it came up. You can see it above.

It will need to be changed. Looks like on that passing commit only labelling CI was run - see https://github.com/freeCodeCamp/freeCodeCamp/runs/3804026331, hence no linting error then.

**Comment by ashiskumar-1999** [comment]
So, Can you tell me what to do so that I can avoid this prettier issue?

**Comment by ashiskumar-1999** [comment]
I have cross-checked every line of code. It seems like there will be no ";" need in the code but I don't know why this error has come?

File: client/src/components/SolutionViewer/ProjectModal.tsx
Comment (ShaunSHamilton) [inline client/src/components/SolutionViewer/ProjectModal.tsx:22]:
```suggestion
  solution: string;
```

File: client/src/components/SolutionViewer/ProjectModal.tsx
Comment (ShaunSHamilton) [inline client/src/components/SolutionViewer/ProjectModal.tsx:31]:
```suggestion
const ProjectModal = ({
  isOpen,
  projectTitle,
  challengeFiles,
  solution,
  handleSolutionModalHide
}: ProjectModalProps): JSX.Element => {
```

File: client/src/components/SolutionViewer/ProjectModal.tsx
Comment (ShaunSHamilton) [inline client/src/components/SolutionViewer/ProjectModal.tsx:51]:
```suggestion
        <SolutionViewer challengeFiles={challengeFiles} solution={solution} />
```

File: client/src/components/SolutionViewer/ProjectModal.tsx
Comment (ShaunSHamilton) [inline client/src/components/SolutionViewer/ProjectModal.tsx:18]: I cannot make a suggestion for all of this file.

Please delete this comment - this migration is what the `TODO` is referring to.

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
@ashiskumar-1999 

Thank you, for taking the time to contribute.

You have not addressed all of the previous suggestions. Please make use of the GitHub UI in the `Files changed` tab to 'batch' all of these suggestions. Then, click the `commit suggestions` button to automatically accept the changes.

---

Also, please edit your PR title to follow the recommended format described in the [Contributor Docs](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request)

And be aware that as this PR has been made from your `main` branch, it will require a reset, once this is merged, before any further contributions can be made.

Hope this helps. If you are stuck on anything, or do not understand, please feel free to reach out to us on either [Chat](https://chat.freecodecamp.org/) or the forum.

File: client/src/components/SolutionViewer/ProjectModal.tsx
Comment (nhcarrigan) [reply client/src/components/SolutionViewer/ProjectModal.tsx:22]: This throws a compiler error in the `show-project-links.tsx` file because the initial state of `solution` is `null`.

Comment (nhcarrigan) [reply client/src/components/SolutionViewer/ProjectModal.tsx:22]: S'why I proposed the `String(solution)` approach.

hacky, and I'm open to a better way to do it, but it would prevent blocking this PR.

**Comment by ashiskumar-1999** [comment]
Hey @ShaunSHamilton, I have added the changes in the files. There is no need of adding those changes again.

**Review by ashiskumar-1999 — COMMENTED** [review]
Thank you for reviewing the errors.

**Comment by ashiskumar-1999** [comment]
Hey @ShaunSHamilton, I don't think that there is any need for changes to those requests because those have been already changed by me. If I won't resolve those conversations then it will throw the error of unresolved conversations. But if I resolve those conversations then the only error came up i.e the "changes requested" error. Can you please tell me how to avoid those conversations?

**Comment by ShaunSHamilton** [comment]
> I don't think that there is any need for changes to those requests because those have been already changed by me.

You have not made the suggested changes in the requests.

Currently, one change is still being discussed. As such, it should not be marked as resolved. For the 3rd time, I have marked the conversation as _unresolved_ . Please do not mark a suggestion/comment as resolved, unless you have taken the suggestion.

> If I won't resolve those conversations then it will throw the error of unresolved conversations.

There is no error being thrown. If you commit the suggestion in a review comment, then GitHub will automatically mark the conversation as resolved. Otherwise, it is most agreeable to let the one who made the comment mark the comment as resolved themselves.

This helps avoid the situation of ignored reviews, and duplicate suggestions.

> Can you please tell me how to avoid those conversations?

I do not understand what you are trying to avoid. This is what needs to happen in this PR:
1) A solution to the currently **unresolved** conversation needs to be decided
2) You may help commit the decided change
3) The original reviewer (me, in this case) needs to mark the conversation as resolved
4) The PR gets reviewed again
5) Once all QA checks are complete, a maintainer can merge the PR

Hope this helps

**Comment by ashiskumar-1999** [comment]
Yes, I'll do it.

**Review by ShaunSHamilton — APPROVED** [review]
Thanks, for doing that 👍 

I just pushed a small change asserting the type for the modal-related components.

---

This LGTM
Otherwise, some more changes will be needed once/when the `Certification.js` file is migrated.

**Review by naomi-lgbt — APPROVED** [review]
Looks good to me! Thanks for all your work on this.

**Pull request merged**
